fiboa_cli/datasets/template.py

```diff
@@ -4,170 +4,143 @@
 # The name of the file will be the name of the converter in the cli.
 # If you name it 'de_abc' you'll be able to run `fiboa convert de_abc` in the cli.
 
-from ..convert_utils import convert as convert_
+from ..convert_utils import BaseConverter
+
+
+class TemplateConverter(BaseConverter):
+    # File(s) to read the data from, usually publicly accessible URLs.
+    # Can read any (zipped) tabular data format that GeoPandas can read through read_file() or read_parquet().
+    # Supported protocols: HTTP(S), GCS, S3, or the local file system
+    #
+    # Multiple options are possible:
+    # 1. a single URL (filename must be in the URL). The file is read as is.
+    sources = "https://fiboa.example/file.xyz"
+
+    # 2. a dictionary with a mapping of URLs (where the filename can't necessarily be determined from the URL) to filenames.
+    # sources = {
+    #   "https://fiboa.example/archive/758?download=1": "us.gpkg"
+    #   "https://fiboa.example/archive/355?download=1": "canada.gpkg"
+    # }
+    # 3. a dictionary with a mapping of URLs to a list of filenames in ZIP ot 7Z files to read from.
+    # sources = {
+    #   "https://fiboa.example/north_america.zip": ["us.gpkg", "canaga.gpkg"]
+    # }
+
+    # If multiple variants are available (e.g. different years), you can replace sources by source_variants.
+    # The dict-key can be used on the cli command line, the value will be used as 'sources'
+    #
+    # source_variants = {
+    #    "2023": "https://fiboa.example/file_2023.xyz"
+    #    "2024": "https://fiboa.example/file_2023.xyz"
+    # }
```

Comment:
I think it could make sense to put the sources into a method which gets passed the variants argument.
Then you can create the sources more easily and don't need to repeat URIs as much...

Like: `return f"https://fiboa.example/file_{variant}.xyz"` or so. You could even do some checks, e.g. check that variant is > 2020 or so when data is only available since 2020.

Reply:
Yes. Although this is explicit. 

You can already do this in `BaseConverter.get_urls()` and just forget about the whole sources/source_variants..